crates/gitbutler-core/src/virtual_branches/integration.rs

```diff
@@ -85,14 +85,28 @@ pub fn get_workspace_head(
     // TODO(mg): Can we make this a constant?
     let committer = get_committer()?;
 
-    // Create merge commit of branch heads.
+    let mut heads: Vec<git2::Commit<'_>> = applied_branches
+        .iter()
+        .filter(|b| b.head != target.sha)
+        .map(|b| repo.find_commit(b.head.into()))
+        .filter_map(Result::ok)
+        .collect();
+
+    if heads.is_empty() {
+        heads = vec![target_commit]
+    }
+
+    // TODO: Why does commit only accept a slice of commits? Feels like we
+    // could make use of AsRef with the right traits.
+    let head_refs: Vec<&git2::Commit<'_>> = heads.iter().collect();
```

Comment:
This clearly is a place where `git2` has a very inconvenient API, as a slice of refs nobody has naturally, ever. It's also a strange choice given that it needs parent-ids, not commit objects.
This goes away with `gitoxide`.